lib/active_job/retry.rb

```diff
@@ -117,6 +125,18 @@ def check_adapter!
       end
     end
 
+    def validate_params
+      if retry_callback && !retry_callback.is_a?(Proc)
```

Comment:
It's a little bit more complicated since the current solution uses `instance_exec` to call the proc, so that `self` is the current instance. So I am not sure if the check is correct for this?